sql/analyzer/indexed_joins.go

```diff
@@ -927,7 +927,13 @@ func addMergeJoins(m *memo.Memo) error {
 						leftColId := getOnlyColumnId(matchedEqFilters[0].filter.Left)
 						rightColId := getOnlyColumnId(matchedEqFilters[0].filter.Right)
 						lIndexScan := makeIndexScan(lIndex, leftColId, lFilters)
+						if lIndexScan == nil {
```

Comment:
maybe a sign that `makeIndexScan` should return a second bool parameter to indicate success/failure so that it is harder for people using this function to accidentally create nil panics

Reply:
Good point. I added a bool return value to the function.